build-tools/packages/build-cli/src/library/markdown.ts

```diff
@@ -41,3 +42,105 @@ export function remarkHeadingLinks(): (tree: Node) => void {
 		});
 	};
 }
+
+/**
+ * A regular expression that extracts an admonition title from a string UNLESS the admonition title is the only thing on
+ * the line.
+ *
+ * Capture group 1 is the admonition type/title. Capture group 2 is any trailing whitespace.
+ *
+ * @remarks
+ *
+ * Description of the regular expression:
+ *
+ * This regular expression matches patterns in the form of `[!WORD]` where WORD can be CAUTION, IMPORTANT, NOTE, TIP, or
+ * WARNING. It ensures that the pattern is not followed by only whitespace characters until the end of the line.
+ * Additionally, it captures any whitespace characters that follow the matched pattern.
+ */
+const ADMONITION_REGEX = /(\[!(?:CAUTION|IMPORTANT|NOTE|TIP|WARNING)])(?!\s*$)(\s*)/gm;
+
+/**
+ * A regular expression to remove single line breaks from text. This is used to remove extraneous line breaks in text
+ * nodes in markdown. This is useful because GitHub sometimes renders single line breaks, and sometimes it ignores them
+ * like the CommonMark spec describes. Removing them ensures that markdown renders as expected across GitHub.
+ */
+const SOFT_BREAK_REGEX = /$[^$]/gms;
+
+/**
+ * A remarkjs/unist plugin that strips soft line breaks. This is a workaround for GitHub's inconsistent markdown
+ * rendering in GitHub Releases. According to CommonMark, Markdown paragraphs are denoted by two line breaks, and single
+ * line breaks should be ignored. But in GitHub releases, single line breaks are rendered. This plugin removes the soft
+ * line breaks so that the markdown is correctly rendered.
+ */
+export function stripSoftBreaks(): (tree: Node) => void {
+	return (tree: Node): void => {
+		// strip soft breaks
+		visit(tree, "text", (node: { value: string }) => {
+			node.value = node.value.replace(SOFT_BREAK_REGEX, " ");
+		});
+
+		// preserve GitHub admonitions; without this the line breaks in the alert are lost and it doesn't render correctly.
+		visit(tree, "blockquote", (node: Node) => {
+			visit(node, "text", (innerNode: { value: string }) => {
+				// If the text is an admonition title, split
+				innerNode.value = innerNode.value.replace(ADMONITION_REGEX, "$1\n");
+			});
+		});
+	};
+}
+
+export function removeSectionContent(options: { heading: string | RegExp }): (
+	tree: Root,
+) => void {
+	return function (tree: Root) {
+		headingRange(tree, options.heading, (start, nodes, end, info) => {
+			console.log(`removing section ${options.heading}`);
+			return [
+				start,
+				// No child nodes - effectively empties the section.
+				end,
+			];
+		});
+	};
+}
+
+export function removeHeadingsAtLevel(options: { level: 1 | 2 | 3 | 4 | 5 | 6 }): (
```

Comment:
Can't quite tell how this is invoked by `remark`, and if the existence of `headingRemoved` could cause that only the first heading of a given level was removed but the rest remained.

Reply:
This was refactored and I think I missed removing the headingRemoved flag. I'll take a fresh look.